e2e/mock/server.go

```diff
@@ -3,39 +3,78 @@ package mock
 import (
 	"net"
 
-	"github.com/hyperledger/fabric-protos-go/orderer"
-	"github.com/hyperledger/fabric-protos-go/peer"
 	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
 )
 
 type Server struct {
-	GrpcServer *grpc.Server
-	Listener   net.Listener
+	peers   []*Peer
+	orderer *Orderer
 }
 
-func (s *Server) Start() {
+func ConstructServer(peerN int, Credentials credentials.TransportCredentials) (server *Server, err error) {
 	blockC := make(chan struct{}, 1000)
-
-	p := &Peer{
-		BlkSize: 10,
-		TxC:     blockC,
+	server = &Server{}
+	server.peers = make([]*Peer, 0)
+	// for peers
+	for i := 0; i < peerN; i++ {
+		lis, err := net.Listen("tcp", "127.0.0.1:0")
+		if err != nil {
+			return nil, err
+		}
+		grpcServer := grpc.NewServer()
+		if Credentials != nil {
+			grpcServer = grpc.NewServer(grpc.Creds(Credentials))
+		}
+		peer := &Peer{
+			Listener:   lis,
+			GrpcServer: grpcServer,
+			BlkSize:    10,
+			TxC:        blockC,
+		}
+		peer.Prepare()
+		server.peers = append(server.peers, peer)
+	}
+	// for orderer
```

Comment:
same here, comment is meant to be read by others, instead of making short notes in code